src/handler/handler.h

```diff
@@ -0,0 +1,16 @@
+#ifndef HANDLER_H
+#define HANDLER_H
+
+#include "http/request.h"
+#include "http/response.h"
+
+// Abstract base Handler for server
+// It createes a Response based on the Request and type of Handler
+class RequestHandler {
+public:
+	// Function for handler to execute when called
+	virtual Response handle(Request request) = 0;
```

Comment:
Just FYI: I changed this API in a later commit to return a boolean, and set up Response as an out-param. The idea is that some handlers may fail and be unable to recover - i.e., a streaming file handler which loses access to its file halfway through the response.